src/main/lrsql/util/reaction.clj

```diff
@@ -101,3 +102,19 @@
   [ruleset-bytes]
   (with-open [r (io/reader ruleset-bytes)]
     (cjson/parse-stream r (partial keyword nil))))
+
+(s/fdef generate-statement
+  :args (s/cat :cond-map (s/map-of ::rs/condition-name
+                         ::rs/val)
+               :statement map?))
+
+(defn generate-statement [cond->statement template]
+  (walk/postwalk #(if (and (map? %)
+                           (= (key (first %)) "$templatePath"))
+                    (let [path (val (first %))
+                          result (get-in cond->statement path :not-found)]
+                      (case result
+                        :not-found (throw (Exception. (str "No value found at " path)))
```

Comment:
Generally by convention we throw `clojure.lang.ExceptionInfo` errors, so we can tack on any relevant data. Check out `clojure.core/ex-info` for that. The convention is we pick a namespaced `:type` key to give them that is descriptive of the error (ex. `::invalid-path`), and then any other stuff like the path itself.